packages/server/tests/acceptance/ws/subscribe.spec.ts

```diff
@@ -20,39 +20,120 @@
 
 // external resources
 import { solidity } from "ethereum-waffle";
-import chai, {expect} from "chai";
+import chai, {assert, expect} from "chai";
 chai.use(solidity);
 
 import {Utils} from '../../helpers/utils';
+import { finished } from "stream";
+import bodyParser from "koa-bodyparser";
 const {ethers} = require('ethers');
 
-describe('@web-socket Acceptance Tests', async function () {
+const FOUR_TWENTY_NINE_RESPONSE = 'Unexpected server response: 429';
+
+const establishConnection = async () => {
+        const provider = await new ethers.providers.WebSocketProvider(
+            `ws://localhost:${process.env.WEB_SOCKET_PORT}`
```

Comment:
Nit: make this a class variable set in the before All maybe.
We use it quite a few times.

Reply:
Done.